bank_lib/src/bank.rs

```diff
@@ -0,0 +1,315 @@
+        }
+        *balance -= amount;
+        let op_index = self.record_operation(Operation::Withdraw {
+            name: name.to_string(),
+            amount,
+        });
+        Ok(op_index)
+    }
+
+    /// Переводит деньги с одного счета на другой.
+    ///
+    /// # Пример
+    ///
+    /// ```
+    /// use bank_lib::Bank;
+    /// let mut bank = Bank::new();
+    /// bank.create_account("Alice".to_string()).unwrap();
+    /// bank.create_account("Bob".to_string()).unwrap();
+    /// bank.deposit("Alice", 100).unwrap();
+    /// assert!(bank.transfer("Alice", "Bob", 50).is_ok());
+    /// ```
+    ///
+    /// # Ошибки
+    ///
+    /// Возвращает `BankError::AccountNotFound`, если один из счетов не существует.
+    /// Возвращает `BankError::InvalidAmount`, если сумма перевода <= 0.
+    /// Возвращает `BankError::SameAccountTransfer`, если перевод на тот же счет.
+    /// Возвращает `BankError::InsufficientFunds`, если на счете отправителя недостаточно средств.
+
+    pub fn transfer(&mut self, from: &str, to: &str, amount: i64) -> Result<usize, BankError> {
+        if amount <= 0 {
+            return Err(BankError::InvalidAmount);
+        }
+        if from == to {
+            return Err(BankError::SameAccountTransfer);
+        }
+        let from_balance = self
+            .accounts
+            .get_mut(from)
+            .ok_or(BankError::AccountNotFound)?;
+        if *from_balance < amount {
+            return Err(BankError::InsufficientFunds);
+        }
+        *from_balance -= amount;
+        let to_balance = self
+            .accounts
+            .get_mut(to)
+            .ok_or(BankError::AccountNotFound)?;
+        *to_balance += amount;
+        let op_index = self.record_operation(Operation::Transfer {
+            from: from.to_string(),
+            to: to.to_string(),
+            amount,
+        });
+        Ok(op_index)
+    }
+
+    /// Возвращает баланс указанного счета.
+    ///
+    /// # Пример
+    ///
+    /// ```
+    /// use bank_lib::Bank;
+    /// let mut bank = Bank::new();
+    /// bank.create_account("Alice".to_string()).unwrap();
+    /// assert_eq!(bank.get_balance("Alice"), Ok(0));
+    /// ```
+    ///
+    /// # Ошибки
+    ///
+    /// Возвращает `BankError::AccountNotFound`, если счет не существует.
+
+    pub fn get_balance(&self, name: &str) -> Result<i64, BankError> {
+        self.accounts
+            .get(name)
+            .cloned()
+            .ok_or(BankError::AccountNotFound)
+    }
+
+    /// Возвращает операцию по индексу.
+    ///
+    /// # Пример
+    ///
+    /// ```
+    /// use bank_lib::{Bank, Operation};
+    /// let mut bank = Bank::new();
+    /// let op_index = bank.create_account("Alice".to_string()).unwrap();
+    /// assert_eq!(bank.get_operation(op_index).unwrap(), &Operation::CreateAccount { name: "Alice".to_string() });
+    /// ```
+
+    pub fn get_operation(&self, index: usize) -> Option<&Operation> {
```

Comment:
```suggestion
    /// ```
    pub fn get_operation(&self, index: usize) -> Option<&Operation> {
```